modular_skyrat/modules/armaments/code/armament_station.dm

```diff
@@ -0,0 +1,209 @@
+			))
+		if(!LAZYLEN(armament_subcategories))
+			continue
+		data["armaments_list"] += list(list(
+			"category" = armament_category,
+			"category_limit" = GLOB.armament_entries[armament_category][CATEGORY_LIMIT],
+			"category_uses" = used_categories[armament_category],
+			"subcategories" = armament_subcategories,
+		))
+
+	return data
+
+/obj/machinery/armament_station/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+
+	switch(action)
+		if("equip_item")
+			var/datum/armament_entry/armament_entry
+			for(var/category in GLOB.armament_entries)
+				for(var/subcategory in GLOB.armament_entries[category][CATEGORY_ENTRY])
+					armament_entry = locate(params["armament_ref"]) in GLOB.armament_entries[category][CATEGORY_ENTRY][subcategory]
+					if(armament_entry)
+						break
+				if(armament_entry)
+					break
+			if(!armament_entry)
+				return
+			if(products && !(armament_entry.type in products))
+				return
+			select_armament(usr, armament_entry)
+		if("eject_card")
+			eject_card(usr)
+
+/obj/machinery/armament_station/proc/eject_card(mob/user)
+	if(!inserted_card)
+		to_chat(user, span_warning("No card inserted!"))
+		return
+	inserted_card.forceMove(drop_location())
+	user.put_in_hands(inserted_card)
+	inserted_card = null
+	to_chat(user, span_notice("Card ejected!"))
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 70)
+
+/obj/machinery/armament_station/proc/select_armament(mob/user, datum/armament_entry/armament_entry)
+	if(!inserted_card)
+		to_chat(user, span_warning("No card inserted!"))
+		return
+	if(used_categories[armament_entry.category] >= GLOB.armament_entries[armament_entry.category][CATEGORY_LIMIT])
+		to_chat(user, span_warning("Category limit reached!"))
+		return
+	if(purchased_items[armament_entry] >= armament_entry.max_purchase)
+		to_chat(user, span_warning("Item limit reached!"))
+		return
+	if(!ishuman(user))
+		return
+	if(!inserted_card.use_points(armament_entry.cost))
+		to_chat(user, span_warning("Not enough points!"))
+		return
+
+	var/mob/living/carbon/human/human_to_equip = user
+
+	var/obj/item/new_item = new armament_entry.item_type(drop_location())
+
+	used_categories[armament_entry.category]++
+	purchased_items[armament_entry]++
+
+	playsound(src, 'sound/machines/machine_vend.ogg', 50, TRUE, extrarange = -3)
+
+	if(armament_entry.equip_to_human(human_to_equip, new_item))
+		to_chat(user, span_notice("Equipped directly to your person."))
+		playsound(src, 'sound/items/equip/toolbelt_equip.ogg', 100)
+	armament_entry.after_equip(drop_location(), new_item)
+
+/**
+ * Armament points card
+ *
+ * To be used with the armaments vendor.
+ */
+/obj/item/armament_points_card
+	name = "armament points card"
+	desc = "A points card that can be used at an Armaments Station or Armaments Dealer."
+	icon = 'modular_skyrat/modules/armaments/icons/armaments.dmi'
+	icon_state = "armament_card"
+	/// How many points does this card have to use at the vendor?
+	var/points = 10
```

Comment:
make the default number a define, maybe?

Reply:
No need, this card has no default, as it's the base type